lib/utils.pm

```diff
@@ -1658,6 +1659,52 @@ sub script_output_retry {
     die("Waiting for Godot: $cmd") if $die;
 }
 
+
+=head2 validate_script_output_retry
+
+ validate_script_output_retry($cmd, $check, [retry => $retry], [delay => $delay], [timeout => $timeout]);
+
+Repeat command until validate_script_output succeeds or die. Return the output of the command on success.
+
+C<$retry> refers to the number of retries and defaults to C<10>.
+
+C<$delay> is the time between retries and defaults to C<30>.
+
+If the command doesn't succeed after C<$retry> retries,
+this function will die.
+
+Example:
+
+ validate_script_output_retry('ping -c1 -W1 machine', m/1 packets transmitted/, retry => 5, delay => 60);
+
+=cut
+sub validate_script_output_retry {
+    my ($cmd, $check) = splice(@_, 0, 2);
+    my %args = testapi::compat_args({retry => 10, delay => 30, timeout => 90}, [], @_);
+    my $retry = delete $args{retry};
+    my $delay = delete $args{delay};
+
+    my $ret;
+
+    my $exec = "timeout $args{timeout} $cmd";
+    # Exclamation mark needs to be moved before the timeout command, if present
+    if (substr($cmd, 0, 1) eq "!") {
+        $cmd = substr($cmd, 1);
+        $cmd =~ s/^\s+//;    # left trim spaces after the exclamation mark
+        $exec = "! timeout $args{timeout} $cmd";
+    }
+
+    $args{timeout} += 3;    # timeout for script_run must be larger than for the 'timeout ...' command
+    for (1 .. $retry) {
+        eval { $ret = validate_script_output($exec, $check, %args, procced_on_failure => 1); };
+        return $ret if (defined($ret));
+        die("Can't validate output after $retry retries") if $retry == $_;
+        record_info("Retry", "validate_script_output timed out. Retrying...");
+        sleep $delay;
+    }
+    return $ret;
```

Comment:
I think the code will never reach this `return` as the loop handles success or fail results.
In such a case, you might move the `die` statement outside the loop, as it means that the loop exhausted all retries.

Reply:
good point, will do so :)